WooCommerce/Classes/ViewRelated/InAppPurchases/InAppPurchasesDebugView.swift

```diff
@@ -1,22 +1,71 @@
 import SwiftUI
 import Yosemite
 
+private enum SiteIDSourceType: String, Equatable, CaseIterable {
+    case appUser
+    case environmentVariable
+
+    var title: String {
+        switch self {
+        case .appUser:
+            return "App user's site"
+        case .environmentVariable:
+            return "Environment variable"
+        }
+    }
+
+    var noSourceIDFoundHint: String {
+        switch self {
+        case .appUser:
+            return "Please make sure that the user has a valid Site ID"
+        case .environmentVariable:
+            return "Please set your Int64 test site id to the Xcode environment variable with name \(Constants.siteIdEnvironmentVariableName)."
+        }
+    }
+
+    func retrieveUpgradingSiteID() -> Int64? {
+        switch self {
+        case .appUser:
+            return ServiceLocator.stores.sessionManager.defaultStoreID
+        case .environmentVariable:
+            guard let stringSiteID = ProcessInfo.processInfo.environment[Constants.siteIdEnvironmentVariableName] else {
+                return nil
+            }
+
+            return Int64(stringSiteID)
+        }
+    }
+}
+
 @MainActor
 struct InAppPurchasesDebugView: View {
     private let inAppPurchasesForWPComPlansManager = InAppPurchasesForWPComPlansManager()
     @State var products: [WPComPlanProduct] = []
     @State var entitledProductIDs: Set<String> = []
     @State var inAppPurchasesAreSupported = true
     @State var isPurchasing = false
+    @State private var selectedSiteIDSourceType: SiteIDSourceType = .appUser
     @State private var purchaseError: PurchaseError? {
         didSet {
             presentAlert = purchaseError != nil
         }
     }
     @State var presentAlert = false
 
+
+
     var body: some View {
         List {
+            Section("Upgrading Site ID Source") {
+                Picker(selection: $selectedSiteIDSourceType, label: EmptyView()) {
+                    ForEach(SiteIDSourceType.allCases, id: \.self) { option in
+                        Text(option.title)
+                                }
+                            }
+                            .pickerStyle(.segmented)
+
+            }
+
```

Comment:
Formatting seems off here, but linter is ok with it 🤷🏻‍♂️ : 
```suggestion
             Section("Upgrading Site ID Source") {
                Picker(selection: $selectedSiteIDSourceType, label: EmptyView()) {
                    ForEach(SiteIDSourceType.allCases, id: \.self) { option in
                        Text(option.title)
                    }
                }
                .pickerStyle(.segmented)
            }
```